assets/src/edit-story/components/header/buttons/update.js

```diff
@@ -55,25 +68,24 @@ function Update() {
     [saveStory, isSaving]
   );
 
-  let text;
-  switch (status) {
-    case 'publish':
-    case 'private':
-      text = __('Update', 'web-stories');
-      break;
-    case 'future':
-      text = __('Schedule', 'web-stories');
-      break;
-    default:
-      text = __('Save draft', 'web-stories');
-      return (
-        <Outline
-          onClick={() => saveStory({ status: 'draft' })}
-          isDisabled={isSaving || isUploading || !hasNewChanges}
-        >
-          {text}
-        </Outline>
-      );
+  const isSavingDraftDisabled =
+    !hasMetaBoxes && (isSaving || isUploading || !hasNewChanges);
+
+  if (!['publish', 'private', 'future'].includes(status)) {
+    return (
+      <Outline
+        onClick={() => saveStory({ status: 'draft' })}
+        isDisabled={isSavingDraftDisabled}
+      >
+        {__('Save draft', 'web-stories')}
+      </Outline>
+    );
+  }
+
+  let text = __('Update', 'web-stories');
+
+  if (status === 'future') {
+    text = __('Schedule', 'web-stories');
```

Comment:
@swissspidy Can you explain change a little. The switch statement before was much easier to read than the way it has been changed.

Reply:
I found it easier this way. Was odd to only return something in the default case, and there were only 2 other cases. Now it simply returns early when possible.

Reply:
Getting rid of multiple if statement to make them more readable is exactly the point of a switch statement. If this refactor is not requires for this functionality it should be removed and handled in another PR.